scripts/make-g-cloud-7-live.py

```diff
@@ -0,0 +1,120 @@
+        if draft_service['status'] == 'submitted' and not supplier.get('serviceId')
+    )
+
+
+def parse_document_url(url, framework_slug):
+    pattern = r'/{}/(\d+)/(\d+)-(.*)$'.format(re.escape(framework_slug))
+    match = re.search(pattern, url)
+    if not match:
+        raise ValueError("Could not parse document URL {}".format(url))
+    return {
+        'supplier_id': match.group(1),
+        'draft_id': match.group(2),
+        'document_name': match.group(3),
+    }
+
+
+def get_draft_document_path(parsed_document, framework_slug):
+    return '{framework_slug}/{supplier_id}/{draft_id}-{document_name}'.format(
+        framework_slug=framework_slug,
+        **parsed_document)
+
+
+def get_live_document_path(parsed_document, framework_slug, service_id):
+    return '{framework_slug}/documents/{supplier_id}/{service_id}-{document_name}'.format(
+        framework_slug=framework_slug,
+        service_id=service_id,
+        **parsed_document)
+
+
+def make_draft_service_live(client, supplier, draft_service, framework_slug):
+    print("  > Migrating draft {} - {}".format(draft_service['id'], draft_service['serviceName']))
+    if DRY_RUN:
+        service_id = random.randint(1000, 10000)
+        print("    > generating random test service ID".format(service_id))
+    else:
+        service = client.publish_draft_service(draft_service['id'], 'make-g-cloud-7-live script')
+        service_id = service['id']
+        print("    > draft service published - new service ID {}".format(service_id))
+
+    for document_key in DOCUMENT_KEYS:
+        if document_key not in draft_service:
+            print("    > Skipping {}".format(document_key))
+        else:
+            parsed_document = parse_document_url(draft_service[document_key], framework_slug)
+            assert_equal(str(parsed_document['supplier_id']), str(supplier['supplierId']))
+            assert_equal(str(parsed_document['draft_id']), str(draft_service['id']))
+
+            draft_document_path = get_draft_document_path(parsed_document, framework_slug)
+            live_document_path = get_live_document_path(parsed_document, framework_slug, service_id)
+            if not DRAFT_BUCKET.path_exists(draft_document_path):
+                raise ValueError(
+                    "Draft document {} does not exist in bucket {}".format(
+                        draft_document_path, DRAFT_BUCKET.bucket_name))
+            message_suffix = '{}:{} to {}:{}'.format(
+                DRAFT_BUCKET.bucket_name, draft_document_path,
+                DOCUMENTS_BUCKET.bucket_name, live_document_path
+            )
+
+            if DRY_RUN:
+                print("    > not copying {}".format(message_suffix))
+            else:
+                DOCUMENTS_BUCKET.bucket.copy_key(live_document_path, DRAFT_BUCKET.bucket_name, draft_document_path)
+                print("    > copied {}".format(message_suffix))
+
```

Comment:
We also need the next step here - to update the new published service with all the new document URLs.  We should probably build a dictionary of the 3/4 new URLs for the draft service and then update all document URLs with one call to `client.update_service` - what do you think?